tool/gravity/cli/gc.go

```diff
@@ -425,13 +425,16 @@ func collectRemoteApplications(operator ops.Operator, clusterKey ops.SiteKey) ([
 	return remoteApps, nil
 }
 
-func validateCanPrunePackages(operator ops.Operator, clusterKey ops.SiteKey) error {
-	operation, progress, err := ops.GetLastOperation(clusterKey, operator)
+func validateCanPrunePackages(operator ops.Operator, cluster ops.Site) error {
+	operation, _, err := ops.GetLastOperation(cluster.Key(), operator)
 	if err != nil {
 		return trace.Wrap(err)
 	}
 
-	if !progress.IsCompleted() {
+	// Use the cluster state to determine the operation progress to account
+	// for older clusters where update operation was not explicitly completed.
+	// TODO(dmitri): remove when there's no more need to support this legacy case
+	if cluster.State != ops.SiteStateActive {
 		return trace.CompareFailed("%v is still in progress. "+
```

Comment:
There may be no operation in progress and cluster can still be not active, e.g. degraded. Then the "xxx is still in progress" message will be confusing. I would just remove parts about active operation from this message.